discord/http.py

```diff
@@ -929,12 +929,120 @@ def send_friend_request(self, username, discriminator):
+        r = Route('DELETE', '/applications/{application_id}/commands/{command_id}',
+                  application_id=application_id, command_id=command_id)
+        return self.request(r)
+
+    def get_guild_application_commands(self, application_id, guild_id):
+        r = Route('GET', '/applications/{application_id}/guilds/{guild_id}/commands',
+                  application_id=application_id, guild_id=guild_id)
+        return self.request(r)
+
+    def create_guild_application_command(self, application_id, guild_id, name, description, options=None):
+        payload = {
+            'name' : name,
+            'description' : description,
+        }
+        if options != None:
+            payload['options'] = options
+        r = Route('POST', '/applications/{application_id}/guilds/{guild_id}/commands',
+                  application_id=application_id, guild_id=guild_id)
+        return self.request(r, json=payload)
+
+    def get_guild_application_command(self, application_id, guild_id, command_id):
+        r = Route('GET', '/applications/{application_id}/guilds/{guild_id}/commands/{command_id}',
+                  application_id=application_id, guild_id=guild_id, command_id=command_id)
+        return self.request(r)
+
+    def edit_guild_application_command(self, application_id, guild_id, command_id, **payload):
+        r = Route('PATCH', '/applications/{application_id}/guilds/{guild_id}/commands/{command_id}',
+                  application_id=application_id, guild_id=guild_id, command_id=command_id)
+        return self.request(r, json=payload)
+
+    def delete_guild_application_command(self, application_id, guild_id, command_id):
+        r = Route('DELETE', '/applications/{application_id}/guilds/{guild_id}/commands/{command_id}',
+                  application_id=application_id, guild_id=guild_id, command_id=command_id)
+        return self.request(r)
+
+    def create_interaction_response(self, interaction_id, interaction_token, response):
+        r = Route('POST', '/interactions/{interaction_id}/{interaction_token}/callback',
+                  interaction_id=interaction_id, interaction_token=interaction_token)
+        return self.request(r, json=response)
+
+    def edit_interaction_response(self, application_id, interaction_token, **payload):
+        r = Route('PATCH', '/webhooks/{application_id}/{interaction_token}/messages/@original',
+                  application_id=application_id, interaction_token=interaction_token)
+        return self.request(r, json=payload)
+
+    def delete_interaction_response(self, application_id, interaction_token):
+        r = Route('DELETE', '/webhooks/{application_id}/{interaction_token}/messages/@original',
+                  application_id=application_id, interaction_token=interaction_token)
+        return self.request(r)
+
+    def create_followup_message(self, application_id, interaction_token, *, file=None, files=None, **payload):
```

Comment:
It is indeed based on `send_files`. The deviation is because it combines `send_message` and `send_files`, and it does the `file` with `files` logic. Seeing the other functions here it would indeed make sense to move the `file` `files` logic to a higher level and maybe split this into two. Is there any reason both `send_message` and `send_files` do not take the `payload` to be the remaining keyword arguments with `**payload`?